freeipa/src/main/java/com/sequenceiq/freeipa/service/operation/OperationService.java

```diff
@@ -97,12 +87,13 @@ public Operation failOperation(String operationId, String failureMessage) {
         return failOperation(operationId, failureMessage, Collections.emptyList(), Collections.emptyList());
     }
 
-    public SyncOperationStatus getSyncOperationStatus(String operationId) {
-        return operationToSyncOperationStatus.convert(getOperationForOperationId(operationId));
-    }
-
-    public OperationStatus getOperationStatus(String operationId, String accountId) {
-        return operationToOperationStatusConverter.convert(getOperationForOperationId(operationId));
+    public Operation getOperationForAccountIdAndOperationId(String accountId, String operationId) {
+        Optional<Operation> operationOptional = operationRepository.findByOperationIdAndAccountId(operationId, accountId);
```

Comment:
So was this functionality was always there but you just didn't use it? What are the use cases for using getOperationForOperationID? should that be removed?

Reply:
yeah. this code was there but not used.

The getter by operation id only is used internally for the "complete" and "fail" methods that transition the operation to either the COMPLETED or FAILED state. These are used internally to the FMS. I didn't want to change the behavior of the callers in this PR. We would have to update the flow CleanupEvent to either hold the account id or retrieve the account id from the Stack prior to completing or failing an operation.

I can do that in a follow-on commit.